tools/nimgrab.nim

```diff
@@ -1,13 +1,20 @@
 import std/[os, httpclient]
 
 proc syncDownload(url, file: string) =
-  var client = newHttpClient()
+  let client = newHttpClient()
   proc onProgressChanged(total, progress, speed: BiggestInt) =
-    echo "Downloading " & url & " " & $(speed div 1000) & "kb/s"
-    echo clamp(int(progress*100 div total), 0, 100), "%"
+    var message = "Downloading "
```

Comment:
Accepting it this time but stop these silly things, the code only became longer without improving performance measurably. The Nim compiler does flatten a nested `&` concat tree.